lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -2861,6 +2861,62 @@ bool FoldChannelShuffle::run(Function *F, const CompilationContext &cctx) {
   return changed;
 }
 
+// Fold Tile -> Add into BatchedAdd wherever applicable.
+bool FoldTileAddIntoBatchedAdd::run(Function *F,
+                                    const CompilationContext &cctx) {
+  LOG_SCOPE(F->getLogContext(), getName());
+
+  bool changed = false;
+  auto &nodes = F->getNodes();
+  for (auto &node : nodes) {
+    auto *addNode = dyn_cast<AddNode>(&node);
+    if (!addNode) {
+      continue;
+    }
+
+    NodeValue batchNode, addedNode;
+    auto &LHS = addNode->getLHS();
+    auto &RHS = addNode->getRHS();
+    auto *tileNode = dyn_cast<TileNode>(LHS);
+
+    // Check if LHS is a tile.
+    if (tileNode) {
```

Comment:
```suggestion
    if (tileNode = dyn_cast<TileNode>(LHS)) {
```
to be consistent with the other cast below. I think this is more "LLVM-esque" anyway.

Reply:
Sure

Reply:
Done!